### ISSUE TITLE
Add AH Bot and Player BOT

### ISSUE THREAD

**Issue by Stianos-hub** [Byloth/cmangos-docker#26]
Hey,

I've gotten everything set up and the game running, but I've been struggling to get AH BOT and Playerbot turned on. Is there an easy way to do this? 

Thanks!

**Comment by Byloth** [comment]
Hi, @Stianos-hub!
Thanks for your interest into this project! 🙃

Actually, yes...
There's a PR opened about it that can't be closed yet.

All the details here: https://github.com/Byloth/cmangos-docker/pull/20

---

I'm not sure about it but, to be precise...
If you're not using the `tbc` version of the server, maybe you can run it up by your own just switching branch and building the Docker images by yourself.

**Comment by Byloth** [comment]
Any help for this PR is greatly appreciated and encouraged! 🥹

**Comment by Stianos-hub** [comment]
Hey, I"ll absolutely take a look. I'm attempting to get this working on classic, but I looking at the code it doesn't look to expansion specific. I see some places I missed some flips, going to try some changes and will post results!

**Comment by Stianos-hub** [comment]
Have you gotten them working on the TBC servers? I've added the changes from the PRs and set it to pull the development branch, but I'm still missing something to get the commands working in game.

**Comment by Byloth** [comment]
> Have you gotten them working on the TBC servers?

No, I think it's still incompatible with the `tbc` version of the server.

TBH, I haven't tested it in the past few days; maybe something has changed.  
I will try further, later this week.

**Comment by Stianos-hub** [comment]
I was working on it this weekend, had to take a break for my real job, but I found that the command changed names for cmake, I was going to try and reincorporate what I learned next chance I get. Biggest things I found were:
The command is actually DBUILD_PLAYERBOTS (With an S!)
![image](https://github.com/user-attachments/assets/e33f175a-8ef9-45e4-b738-d520f478db31)

Running with a relative file path causes issues, and it's better to run it from the working directory.
![image](https://github.com/user-attachments/assets/ed2d3bf8-4d4d-4788-a782-50a28265fc50)

That plus making sure the FullInstallDb.sh runs the commands to add the playerbot tables was what got it working for me on classic. 

Once I find time I wanted to back incorporate this into the docker files and see if I can get it to work. 

Hopefully some of this helps if you want to try and incorporate it into tbc before I get a chance to make some changes

**Comment by Byloth** [comment]
Yeah... No worries! 😉

I would say that's already like this:
1. https://github.com/Byloth/cmangos-docker/blob/f425e4707e2d3f2e9a877253cc7058cf48cb51ef/Dockerfile#L111
2. https://github.com/Byloth/cmangos-docker/blob/82125d95347ae6e516fa99323eb861f692b705bd/runner/entrypoint.sh#L97

**Comment by Byloth** [comment]
Take a look also to this conversation...

https://discord.com/channels/362206349339262976/1280194808048259102/1283185344631668847

**Comment by Byloth** [comment]
@draew6: I just tested playerbots on `tbc` (using the [`feature/bots`](https://github.com/Byloth/cmangos-docker/tree/feature/bots) branch) and it worked fine.

**Comment by draew6** [comment]
@Byloth I changed branch and did docker compose up
 
  ✘ mangosd Error pull access denied for byloth/cmangos-tbc, repository does not exist or may require 'docker login'                                                                                             1.2s 
 ✘ realmd Error  context canceled

**Comment by Byloth** [comment]
Yeah, sorry... My bad!
Since this branch isn't production ready, no images are built automatically.

You have to build it by yourself.

Use...

```
./cmangos-build --help
```

... for more details! 😉

**Comment by draew6** [comment]
@Byloth I had to build 2 images  and it took long time but it was worth it! it's working and bots are living their dream :D didnt test anything specific yet, just interacted with them for 30 minutes  but I think any other problems with them would be out of scope of this project so I think it works as expected from my side! thanks for your contribution and instructions (y)

**Comment by Byloth** [comment]
Awesome, @draew6! 🤩

---

> [...] it took long time but it was worth it!

Yeah... CMaNGOS takes a lot of time to build! 🥲
Using multiple threads, accordingly to your hardware, might help!

---

> [...] but I think any other problems with them would be out of scope of this project [...]

Not necessarily, actually... 🤔  
May I ask you if you can report [here](https://github.com/Byloth/cmangos-docker/pull/20) or [here](https://github.com/Byloth/cmangos-docker/discussions/32) any future issues or additional steps / configurations you might need in the future to improve your experience?

This could also help improve the future documentation about bots, configuration process and so on...

Thanks!

**Comment by Byloth** [comment]
I'm going to lock this conversation.  
All the information you need is already here, IMHO.
  
From now on, please refer to the [PR](https://github.com/Byloth/cmangos-docker/pull/20) for the status of "Bot support" and [Discussion](https://github.com/Byloth/cmangos-docker/discussions/32) if you have any questions or concerns.

Thank you all for your interest in this project! ❤️
